test/test_dataloader.py

```diff
@@ -2222,21 +2221,23 @@ def test_excessive_thread_creation_warning(self):
                 r"excessive worker creation might get DataLoader running slow or even freeze"):
             dataloader = DataLoader(self.dataset, batch_size=2, num_workers=1000)
 
+# Define a global function for testing purposes since local functions cannot be pickled
```

Comment:
Why is that only happening on our arm CI?

Reply:
For context, Python's multiprocessing lib uses the default pickle which does not support pickling non-global functions (this is why this test is only enabled when dill is available, which does support pickling such functions).

This will happen all all Mac OS X devices because multiprocessing defaults to spawning (not forking) on this platform:
https://github.com/python/cpython/blob/30610d28374f5a9698d456cebf3ae496ac01af51/Lib/multiprocessing/context.py#L322-L327

Pickling is necessary in the spawn path to pass the function and arguments to execute, but not in the fork path because fork duplicates the whole memory space already.